### PR TITLE
OPENEUROPA-1695: Implement footer block.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

Implement Footer block in the new oe_corporate_blocks that outputs the raw links using a new theme hook

1. Structure like the ECL pattern
2. Only EN
3. No site identity part
4. The links are the same as on the INFO site
Assert the rendering. You can use behat for this since we do the rendering test in the pattern.
### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: tests/Behat/CorporateBlocksContext.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Defines step which used for behat test of Corporate blocks.
+ */
+class CorporateBlocksContext extends RawDrupalContext {
+
+  /**
+   * Assertion of links in region.
+   *
+   * @Then Links in the :region region should be correct:
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply tests/Behat/CorporateBlocksContext.php]: Fixed.

File: config/blocks/footer.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+corporate_footer:
```
Comment (ademarco) [inline config/blocks/footer.yml]: So we discussed with @sergepavle and we could turn this into a service parameter, to be injected into the block, so that we can override it in the theme and test the block with a fixture, instead of the data contained by the module. This will allow us to update links freely in the module without breaking tests on the theme.

Comment (sergepavle) [reply config/blocks/footer.yml]: Done

File: composer.json [hunk 3]
```diff
@@ -14,16 +14,15 @@
...
+        "openeuropa/behat-transformation-context" : "~0.1",
         "openeuropa/code-review": "~1.0@beta",
         "openeuropa/drupal-core-require-dev": "^8.6",
-        "openeuropa/task-runner": "~1.0-beta4",
+        "openeuropa/task-runner": "~1.0-beta5",
         "phpunit/phpunit": "~6.0"
     },
     "scripts": {
-        "post-create-project-cmd": "./build-module.sh",
-        "drupal-scaffold": "DrupalComposer\\DrupalScaffold\\Plugin::scaffold",
```
Comment (upchuk) [inline composer.json]: Why are these removed. They are both in the module template: https://github.com/openeuropa/drupal-module-template/blob/master/composer.json

@ademarco @voidtek @dxvargas should these be here ?

File: composer.json [hunk 4]
```diff
@@ -14,16 +14,15 @@
         "drupal-composer/drupal-scaffold": "~2.2",
         "drupal/config_devel": "~1.2",
         "drupal/console": "~1.0",
-        "drupal/drupal-extension": "~4.0",
+        "drupal/drupal-extension": "~4.0.0@beta",
```
Comment (upchuk) [inline composer.json]: Why the change in behat extention? 

https://github.com/openeuropa/drupal-module-template/blob/master/composer.json#L17

File: oe_corporate_blocks.module [hunk 5]
```diff
@@ -6,3 +6,66 @@
  */
 
 declare(strict_types = 1);
+
+/**
+ * Implements hook_theme().
+ */
+function oe_corporate_blocks_theme() {
+  return [
+    'corporate_blocks_footer' => [
```
Comment (ademarco) [inline oe_corporate_blocks.module]: On components we should prefix all our theme entries with `oe_`, I suggest to change this into `oe_corporate_footer`.

Comment (sergepavle) [reply oe_corporate_blocks.module]: renamed

File: tests/features/footer-block.feature [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      | EU institutions                         | https://europa.eu/european-union/about-eu/institutions-bodies_en            |
+      | European Union                          | https://europa.eu/european-union/index_en                                   |
+      | About the Commission's new web presence | https://ec.europa.eu/info/about-commissions-new-web-presence_en             |
+      | Language policy                         | https://ec.europa.eu/info/language-policy_en                                |
+      | Resources for partners                  | https://ec.europa.eu/info/resources-partners_en                             |
+      | Cookies                                 | https://ec.europa.eu/info/cookies_en                                        |
+      | Privacy policy                          | https://ec.europa.eu/info/privacy-policy_en                                 |
+      | Legal notice                            | https://ec.europa.eu/info/legal-notice_en                                   |
+      | Contact                                 | https://ec.europa.eu/info/contact_en                                        |
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply tests/features/footer-block.feature]: Done

File: src/Plugin/Block/FooterBlock.php [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    NestedArray::setValue($build, ['#corporate_footer', 'social_media', 'items'], $config->get('social_media_links'));
+
+    NestedArray::setValue($build, ['#corporate_footer', 'about_eu', 'title'], $config->get('about_eu_title'));
+    NestedArray::setValue($build, ['#corporate_footer', 'about_eu', 'items'], $config->get('about_eu_links'));
+
+    NestedArray::setValue($build, ['#corporate_footer', 'bottom_links'], $config->get('bottom_links'));
+
+    // By some reasons drush config import do not do invalidating of cache tags.
+    // So for deployment we should take care about this case and
+    // do cache invalidating by tag "config:oe_corporate_blocks.data.footer".
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply src/Plugin/Block/FooterBlock.php]: fixed

File: src/Plugin/Block/FooterBlock.php [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+    NestedArray::setValue($build, ['#corporate_footer', 'social_media', 'title'], $config->get('social_media_title'));
+    NestedArray::setValue($build, ['#corporate_footer', 'social_media', 'items'], $config->get('social_media_links'));
+
+    NestedArray::setValue($build, ['#corporate_footer', 'about_eu', 'title'], $config->get('about_eu_title'));
+    NestedArray::setValue($build, ['#corporate_footer', 'about_eu', 'items'], $config->get('about_eu_links'));
+
+    NestedArray::setValue($build, ['#corporate_footer', 'bottom_links'], $config->get('bottom_links'));
+
+    // By some reasons drush config import do not do invalidating of cache tags.
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply src/Plugin/Block/FooterBlock.php]: fixed

File: src/Plugin/Block/FooterBlock.php [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+    NestedArray::setValue($build, ['#corporate_footer', 'social_media', 'title'], $config->get('social_media_title'));
+    NestedArray::setValue($build, ['#corporate_footer', 'social_media', 'items'], $config->get('social_media_links'));
+
+    NestedArray::setValue($build, ['#corporate_footer', 'about_eu', 'title'], $config->get('about_eu_title'));
+    NestedArray::setValue($build, ['#corporate_footer', 'about_eu', 'items'], $config->get('about_eu_links'));
+
+    NestedArray::setValue($build, ['#corporate_footer', 'bottom_links'], $config->get('bottom_links'));
+
+    return $build;
```
[the remark this answers is not in the document]
Comment (sergepavle) [reply src/Plugin/Block/FooterBlock.php]: Added
